provider/gce/google/raw.go

```diff
@@ -375,28 +376,36 @@ var doOpCall = func(call opDoer) (*compute.Operation, error) {
 // waitOperation waits for the provided operation to reach the "done"
 // status. It follows the given attempt strategy (e.g. wait time between
 // attempts) and may time out.
-//
-// TODO(katco): 2016-08-09: lp:1611427
-func (rc *rawConn) waitOperation(projectID string, op *compute.Operation, attempts utils.AttemptStrategy, f handleOperationErrors) error {
+func (rc *rawConn) waitOperation(projectID string, op *compute.Operation, retryStrategy retry.CallArgs, f handleOperationErrors) error {
 	// TODO(perrito666) 2016-05-02 lp:1558657
 	started := time.Now()
 	logger.Infof("GCE operation %q, waiting...", op.Name)
-	for a := attempts.Start(); a.Next(); {
+
+	retryStrategy.IsFatalError = func(err error) bool {
+		return !errors.IsNotProvisioned(err)
+	}
+	retryStrategy.Func = func() error {
 		if op.Status == StatusDone {
-			break
+			return nil
 		}
-
 		var err error
 		op, err = rc.checkOperation(projectID, op)
 		if err != nil {
-			return errors.Trace(err)
+			return err
 		}
+		return errors.NewNotProvisioned(nil, "GCE operation not done yet")
 	}
-	if op.Status != StatusDone {
```

Comment:
Is there ever a chance we could be done, but the op status isn't done?

Reply:
There is a fix here I've spotted, but not certain it addresses this fully. Lmk